shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -356,6 +361,46 @@ - (BOOL)launchEngineInternalWithAssetsPath:(NSURL*)assets
   flutterArguments.command_line_argv = argv;
   flutterArguments.platform_message_callback = (FlutterPlatformMessageCallback)OnPlatformMessage;
 
+  // Provide data for release mode engine.
+#if NDEBUG
+  const char* vmSnapshotDataPath = [[[NSBundle mainBundle] pathForResource:kVmSnapshotDataPath
+                                                                    ofType:nil] UTF8String];
+  const char* vmSnapshotInstructionsPath =
+      [[[NSBundle mainBundle] pathForResource:kVmSnapshotInstructionsPath ofType:nil] UTF8String];
+  const char* isolateSnapshotDataPath =
+      [[[NSBundle mainBundle] pathForResource:kIsolateSnapshotDataPath ofType:nil] UTF8String];
+  const char* isolateSnapshotInstructionsPath =
+      [[[NSBundle mainBundle] pathForResource:kIsolateSnapshotInstructionsPath
+                                       ofType:nil] UTF8String];
+  size_t vmSnapshotDataLength;
+  void* vmSnapshotData;
```

Comment:
All these buffers are leaked. Thy need to be unmapped. Consider using a unique pointer with a custom deleter that unmaps the buffer (after the engine is shut down).